src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -77,39 +108,51 @@ export default class ReplyField extends React.Component<Props, State> {
 
         const range = selection.getRangeAt(0);
         const textNode = range.endContainer;
+        const currentCursor = range.endOffset;
+
+        const { mentionString, mentionTrigger } = activeMention;
+        const offset = mentionString.length + mentionTrigger.length;
+        const mentionStart = currentCursor - offset;
 
-        range.setStart(textNode, activeMention.start);
-        range.setEnd(textNode, activeMention.start);
+        range.setStart(textNode, mentionStart);
+        range.setEnd(textNode, mentionStart);
 
         const mentionRect = range.getBoundingClientRect();
 
-        // restore selection
-        range.setStart(textNode, activeMention.end);
-        range.setEnd(textNode, activeMention.end);
+        // restore cursor position
+        range.setStart(textNode, currentCursor);
+        range.setEnd(textNode, currentCursor);
 
         return {
             getBoundingClientRect: () => mentionRect,
         };
     };
 
     handleChange = (nextEditorState: EditorState): void => {
-        const { onChange, onMention } = this.props;
+        const { onChange } = this.props;
 
         onChange(nextEditorState.getCurrentContent().getPlainText());
 
         const activeMention = getActiveMentionForEditorState(nextEditorState);
+        const popupItems = this.getItemList(activeMention ? activeMention.mentionString : '');
```

Comment:
Can this be derived in `render` and `handleKeyDown` rather than being stored in state?